bokeh/models/widgets/widget.py

```diff
@@ -17,3 +17,15 @@ class Widget(LayoutDOM):
     ''' A base class for all interactive widget types.
 
     '''
+
+class NeedsJQuery(object):
+    """ A mixin for widgets requiring JQuery and/or JQuery UI. """
+
+    __javascript__ = [
+        "https://code.jquery.com/jquery-3.2.1.min.js",
+        "https://code.jquery.com/ui/1.12.1/jquery-ui.min.js",
+    ]
+
+    __css__ = [
+        "https://code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css",
+    ]
```

Comment:
> re-implement widgets without jq/jq-ui at all

I would prefer that. In fact, most important widgets are already jquery independent. Most notably sliders and data table depend on jquery(ui) in this PR. Sliders are already reimplemented in PR #5774. This leaves us with data table. Additionally there is date picker and auto complete input, where the former needs some work to make it jquery independent (there's an issue open for that) and the later is pretty trivial to reimplement.